src/index.ts

```diff
@@ -28,9 +28,21 @@ async function main() {
 
   if (stampsConfig) {
     logger.debug('stamps config', stampsConfig)
-    const stampManager = new StampsManager()
-    logger.info('starting postage stamp manager')
-    stampManager.start(stampsConfig)
+    let stampManager: HardcodedStampsManager | AutoBuyStampsManager | ExtendsStampManager
+    const { mode } = stampsConfig
+
+    if (mode === 'hardcoded') {
+      stampManager = new HardcodedStampsManager()
+
+      logger.info('starting hardcoded postage stamp manager')
+      stampManager.start(stampsConfig)
+    } else if (mode === 'autobuy') {
+      logger.info('starting autobuy postage stamp manager')
+      stampManager = new AutoBuyStampsManager(stampsConfig)
+    } else {
+      logger.info('starting extends postage stamp manager')
+      stampManager = new ExtendsStampManager(stampsConfig)
+    }
```

Comment:
I'd suggest to remove the start call from the constructors, and call start on stampManager once here once it is created.

Reply:
I did it this way because `hardcoded start` is entirely different than `autobuy` and `extends`. Hardcoded doesn't use `interval refresh` and I wanted to keep them both separated to avoid combining 2 different approaches. 

I'll modified the start to take care of the narrowing of both modes (interval and interval-less)